tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/homescreen_settings.py

```diff
@@ -5,35 +5,65 @@
 from marionette_driver import expected, By, Wait
 
 from gaiatest.apps.base import Base
-
+from gaiatest.apps.system.regions.activities import Activities
 
 class HomescreenSettings(Base):
 
-    _page_locator = (By.ID, 'homescreen')
-    _icon_layout_locator = (By.CSS_SELECTOR, '#homescreen div.icon-dialog')
-    _icon_layout_confirmation_button_locator = (By.CLASS_NAME, "value-option-confirm")
+    _page_locator = (By.ID, 'homescreens')
+    _change_homescreen_page_locator = (By.ID, 'homescreens-list')
+    _change_homescreen_locator = (By.CSS_SELECTOR, '[data-l10n-id="change-home-screen"]')
+    _get_more_homescreen_locator = (By.CSS_SELECTOR, '[data-l10n-id="get-more-home-screens"]')
+
+    _wallpaper_preview_locator = (By.CSS_SELECTOR, '.wallpaper-preview')
+    _wallpaper_pick_locator = (By.CLASS_NAME, 'wallpaper-button')
+    _pick_cancel_button_locator = (By.CSS_SELECTOR,
+                                             '#action-menu-list > button:nth-child(4)')
```

Comment:
The 4th button in #action-menu-list looks fragile to me. Can't you use #action-menu-list > button[data-l10n-id="cancel"] here?